### PR TITLE
update `devdeps.py` to including the missing dependencies

### LINKED ISSUE #4824: devdeps.py doesn't check for test dependencies
Was just working @stvn66 to get tests running locally, and when running `devdeps.py` it didn't report that flake8 and boto weren't installed which are needed by the tests to run.

I may have been directing @stvn66 incorrectly though.

ping @damianavila

### PR DESCRIPTION
**Pull request by stvn66**
fixes #4824 
Added `flake8` and `boto` as well as the base dependencies required for bokeh: 
- `backports_abc`,
- `jinja2`,
- `markupsafe`,
- `mkl`, (pending correct import call)
- `bumpy`,
- `dateutil`,
- `pyyaml`, (pending correct import call)
- `requests`,
- `six`,
- `tornado`,
- `yaml`

- [ ] issues: fixes #4824 
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

File: scripts/devdeps.py [hunk 1]
```diff
@@ -76,3 +82,18 @@ def depend_check(deps_name, *args):
         'pygments',
     ]
     depend_check('Docs', *docs_deps)
+
+    base_deps = [
+        'backports_abc',
+        'jinja2',
+        'markupsafe',
+        # 'mkl',
```
Comment (bryevdv) [inline scripts/devdeps.py]: this is definitely not required

File: scripts/devdeps.py [hunk 2]
```diff
@@ -76,3 +82,18 @@ def depend_check(deps_name, *args):
...
+        'numpy',
+        'dateutil',
+        # 'pyyaml',
```
Comment (bryevdv) [inline scripts/devdeps.py]: this I think is required?

File: scripts/devdeps.py [hunk 3]
```diff
@@ -67,7 +71,9 @@ def depend_check(deps_name, *args):
         'pytest_cov',
         'pytest_selenium',
         'mock',
-        'websocket'
+        'websocket',
```
Comment (bryevdv) [inline scripts/devdeps.py]: I don't think this is required anymore, just `websocket-client` for testing

Comment (bryevdv) [inline scripts/devdeps.py]: I'm not sure this is used anymore. I think the tests all use `tornado.websocket` now

File: scripts/devdeps.py [hunk 4]
```diff
@@ -58,6 +59,9 @@ def depend_check(deps_name, *args):
         'websocket': 'websocket-client',
         'pytest_selenium': 'pytest-selenium',
         'pytest_cov': 'pytest-cov',
+        # 'mkl': 'mkl',
```
Comment (bryevdv) [inline scripts/devdeps.py]: mkl is definitely not a requirement, it is a completely optional add on for numpy

**Comment by stvn66** [comment]
Some of these may be legacy from what is in the conda package manager.  I determined which dependencies to add by creating a new python 3.4 environment then installing bokeh.  Here is what I ran:
```
➜  book: scripts/> git:(stvn66_4824_update_devdeps.py) conda create --name bokeh_dependencies python=3.4
Fetching package metadata .........
Solving package specifications: ..........

Package plan for installation in environment /opt/intel/intelpython35/envs/bokeh_dependencies:

The following NEW packages will be INSTALLED:

    openssl:    1.0.2h-1
    pip:        8.1.2-py34_0
    python:     3.4.5-0
    readline:   6.2-2
    setuptools: 25.1.6-py34_0
    sqlite:     3.13.0-0
    tk:         8.5.18-0
    wheel:      0.29.0-py34_0
    xz:         5.2.2-0
    zlib:       1.2.8-3

Proceed ([y]/n)? y

Linking packages ...
[      COMPLETE      ]|###########################################################################################################| 100%
No psutil available.
To proceed, please conda install psutil#
# To activate this environment, use:
# $ source activate bokeh_dependencies
#
# To deactivate this environment, use:
# $ source deactivate
#
➜  book: scripts/> git:(stvn66_4824_update_devdeps.py) source activate bokeh_dependencies
(bokeh_dependencies) ➜  book: scripts/> git:(stvn66_4824_update_devdeps.py) conda install bokeh
Fetching package metadata .........
Solving package specifications: ..........

Package plan for installation in environment /opt/intel/intelpython35/envs/bokeh_dependencies:

The following NEW packages will be INSTALLED:

    backports_abc:   0.4-py34_0
    bokeh:           0.12.1-py34_0
    jinja2:          2.8-py34_1
    markupsafe:      0.23-py34_2
    mkl:             11.3.3-0
    numpy:           1.11.1-py34_0
    python-dateutil: 2.5.3-py34_0
    pyyaml:          3.11-py34_4
    requests:        2.10.0-py34_0
    six:             1.10.0-py34_0
    tornado:         4.4.1-py34_0
    yaml:            0.1.6-0

Proceed ([y]/n)? n
```

According to conda, it wants:
-    backports_abc:   0.4-py34_0 
-    jinja2:          2.8-py34_1
-    markupsafe:      0.23-py34_2
-    mkl:             11.3.3-0
-    numpy:           1.11.1-py34_0
-    python-dateutil: 2.5.3-py34_0
-    pyyaml:          3.11-py34_4
-    requests:        2.10.0-py34_0
-    six:             1.10.0-py34_0
-    tornado:         4.4.1-py34_0
-    yaml:            0.1.6-0

I am fine removing them here if they are definitely legacy.  They should probably also be removed from conda package manager.

**Comment by stvn66** [comment]
When I install Bokeh via pip it also installs:
- six
- requests
- PyYAML
- python-dateutil
- Jinja2
- numpy
- tornado

So there were several packages that conda installs that pip does not.

Here are the pip install details:
```
(bokeh_dependencies) ➜  book: ~/> pip install bokeh
zsh: correct 'bokeh' to '.bokeh' [nyae]? n
Collecting bokeh
  Downloading bokeh-0.12.1.tar.gz (3.6MB)
    100% |████████████████████████████████| 3.6MB 198kB/s
Collecting six>=1.5.2 (from bokeh)
  Downloading six-1.10.0-py2.py3-none-any.whl
Collecting requests>=1.2.3 (from bokeh)
  Using cached requests-2.11.0-py2.py3-none-any.whl
Collecting PyYAML>=3.10 (from bokeh)
  Downloading PyYAML-3.11.zip (371kB)
    100% |████████████████████████████████| 378kB 2.1MB/s
Collecting python-dateutil>=2.1 (from bokeh)
  Downloading python_dateutil-2.5.3-py2.py3-none-any.whl (201kB)
    100% |████████████████████████████████| 204kB 3.3MB/s
Collecting Jinja2>=2.7 (from bokeh)
  Downloading Jinja2-2.8-py2.py3-none-any.whl (263kB)
    100% |████████████████████████████████| 266kB 3.8MB/s
Collecting numpy>=1.7.1 (from bokeh)
  Downloading numpy-1.11.1-cp34-cp34m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (3.8MB)
    100% |████████████████████████████████| 3.9MB 198kB/s
Collecting tornado>=4.3 (from bokeh)
  Downloading tornado-4.4.1.tar.gz (456kB)
    100% |████████████████████████████████| 460kB 1.6MB/s
Collecting MarkupSafe (from Jinja2>=2.7->bokeh)
  Downloading MarkupSafe-0.23.tar.gz
Collecting backports_abc>=0.4 (from tornado>=4.3->bokeh)
  Downloading backports_abc-0.4-py2.py3-none-any.whl
Building wheels for collected packages: bokeh, PyYAML, tornado, MarkupSafe
  Running setup.py bdist_wheel for bokeh ... done
  Stored in directory: /Users/schowell/Library/Caches/pip/wheels/43/81/31/65c9854222564898592b00d984fd49c60fa8de2751236622f0
  Running setup.py bdist_wheel for PyYAML ... done
  Stored in directory: /Users/schowell/Library/Caches/pip/wheels/4a/bf/14/d79994d19a59d4f73efdafb8682961f582d45ed6b459420346
  Running setup.py bdist_wheel for tornado ... done
  Stored in directory: /Users/schowell/Library/Caches/pip/wheels/c2/65/99/727ea73b981e0ef1fb8ab678ecf86f39c809230f5301ac3a40
  Running setup.py bdist_wheel for MarkupSafe ... done
  Stored in directory: /Users/schowell/Library/Caches/pip/wheels/a3/fa/dc/0198eed9ad95489b8a4f45d14dd5d2aee3f8984e46862c5748
Successfully built bokeh PyYAML tornado MarkupSafe
Installing collected packages: six, requests, PyYAML, python-dateutil, MarkupSafe, Jinja2, numpy, backports-abc, tornado, bokeh
Successfully installed Jinja2-2.8 MarkupSafe-0.23 PyYAML-3.11 backports-abc-0.4 bokeh-0.12.1 numpy-1.11.1 python-dateutil-2.5.3 requests-2.11.0 six-1.10.0 tornado-4.4.1
(bokeh_dependencies) ➜  book: ~/>
```

**Comment by stvn66** [comment]
@bryevdv, if you give me a list of packages that should be required, I will update this PR.

**Comment by bryevdv** [comment]
I think it's the pip list. conda also installs the C yaml lib (pip does not do that) and conda also install MKL optimizations (pip also does not do that). 

The only one I am not sure about is `backports_abc` I have no idea what that is. Maybe some 2.7 compat thing?

**Comment by stvn66** [comment]
Just looking at this again.  When I run `conda install jinja2` it also brings in `markupsafe`.  Similarly, when I run `conda install numpy` it brings in `mkl`.

File: scripts/devdeps.py (hunk 3 above)
Comment (stvn66) [inline scripts/devdeps.py]: I did not add `websocket`, just the comma.  Should I remove `websocket`?

**Comment by stvn66** [comment]
I updated the file to remove the dependencies of dependencies.  The base dependencies are now set to:
-        'jinja2',
-        'numpy',
-        'dateutil',
-        'requests',
-        'tornado',
-        'yaml',

File: scripts/devdeps.py (hunk 3 above)
Comment (bryevdv) [inline scripts/devdeps.py]: It actually maps to `websocket-client` which I do have installed, so maybe it is still used in the test. Let's keep it for now

**Comment by bryevdv** [comment]
Thanks @stvn66 !

**Pull request merged**